### ISSUE TITLE
3.20 RightClickAPI is broken

### ISSUE THREAD

**Issue by FormBurden** [Vectorial1024/v1024_civilian_fleets#11]
In public beta 3.20 that released today the RightClickAPI has been broken due to the fact that EgoSoft has added some right-click features. So your mod doesn't work anymore on the beta, which will probably be the same when released. I tried contacting the author for RIghtClickAPI but nothing thus far. Hope gets a fix soon. I have been following your Mod for a while because it's really handy with cleaning up the UI with fleet management and such. And you've mentioned to completely getting rid of RightClickAPI and going SirNukesAPI. I'm guessing that SirNuke's Mod has right-click features or something? So this Mod can create fleets and such. Or are you completely re-writing how it works?

**Comment by Vectorial1024** [comment]
Oh, so it broke again...

An analogy for this is that, for example, I want to get something inside a room, and there are two doors to that room: an unstable one (Right Click API), and a not-so-known one (SirNukes API). Essentially if I prepare myself good enough then I am able to decide e.g. I want to use this door or that door to enter the room.

It's just that I have adopted Right Click API for its nicer UI look, and I don't think SirNukes API can offer the same nice look for the buttons (at least at this moment), so I will need some time to get over it.

**Comment by Vectorial1024** [comment]
Update: at the moment, if I use SirNukes' API, I will know which ship will receive civilian subordinates, but I will never know which ships will join the civilian commander, so as a result, I cannot initiate civilian fleets using SirNukes' API.

So, at this moment, the usual workaround, you may want to go to contents.xml, and delete the `<dependency id="RightClickAPI" />` part of the code, and disable Right Click API, so that this mod can start in Fallback Mode. Hopefully things are stable in fallback mode to be reliable.

**Comment by FormBurden** [comment]
I see. All is well, RightClickAPI is a great feature. I'm barely getting into Modding myself but I've seen some mod authors state that SirNukes API is superior to RightClickAPI and the UI one from same author, but then again, those folks don't even use any kind of right-click mechanics, so idk what they're talking about. 

Anyway, I'll edit the dependency out, never thought of that. So thanks. Hopefully RightClickAPI gets fixed, but it seems the author of that mod doesn't get on Nexus that often. Sad face...

**Comment by Vectorial1024** [comment]
Right Click API is some fancy but fragile stuff, while SirNukes' API is some stable but simple stuff. They got different target uses, so I use them both.

Regarding Right Click API, I noticed a pattern, where the current maintainer would disappear until EgoSoft finalize a beta version and put it to general release, and then that guy would come out and fix the Right Click API. After that, they would disappear again even e.g. EgoSoft made a new beta, and wait until the next official release.

I would wait until EgoSoft finalize their beta before updating things but then people keep using the beta versions. It was a mess when people complained about this mod breaking for v3.0 beta, so I decided to make a Fallback Mode for them and for future beta versions.
